src/PublicApiAnalyzers/UnitTests/DeclarePublicAPIAnalyzerTestsBase.cs

```diff
@@ -3219,6 +3224,77 @@ public async Task Record_ImplicitProperty_FixAsync()
             await VerifyNet50CSharpAdditionalFileFixAsync(source, shippedText, unshippedText, fixedUnshippedText);
         }
 
+        [Fact]
+        [WorkItem(6759, "https://github.com/dotnet/roslyn-analyzers/issues/6759")]
+        public async Task TestExperimentalApiAsync()
+        {
+            var source = $$"""
+                using System.Diagnostics.CodeAnalysis;
+
+                [Experimental("ID1")]
+                {{EnabledModifierCSharp}} class {|{{AddNewApiId}}:{|{{AddNewApiId}}:C|}|}
+                {
+                }
+                """;
+
+            var shippedText = @"";
+            var unshippedText = @"";
+            var fixedUnshippedText = @"[ID1]C
+[ID1]C.C() -> void";
+
+            await VerifyNet80CSharpAdditionalFileFixAsync(source, shippedText, unshippedText, fixedUnshippedText);
+        }
```

Comment:
Consider also testing removal of Experimental attribute (shipped file contains the experiment, but API no longer marked as experimental) and change of Experimental attribute (however I'm not sure it's a realistic/common scenario)

Reply:
Currently this will be treated as simply removing an old API and adding a new one. We can work to improve this experience in the future based on how things go with Roslyn experiments.